### PR TITLE
feat: more robust binary monitoring

### PR DESCRIPTION
**Pull request by fluffypony**
**Description**
---
Implements a comprehensive binary retry mechanism to make Tari Universe significantly more robust against subprocess startup failures. This enhancement automatically retries failed binaries up to 10 times (configurable) before requiring user intervention, eliminating the need for full application restarts due to temporary binary failures.

**Motivation and Context**
---
Previously, when any subprocess (minotari_node, xmrig, gpu_miner, tor, p2pool, etc.) failed to start, Universe would require a complete application restart. This created a poor user experience, especially in environments with temporary network connectivity issues, resource contention during system startup, antivirus interference, or permission delays.

The new retry mechanism distinguishes between:
1. **Initial startup failures** - retries up to 10 times with 5-second delays
2. **Runtime crashes** - retries up to 3 times with 10-second delays for previously healthy processes

**How Has This Been Tested?**
---
- **Startup Failure Simulation**: Modified binary paths to trigger startup failures and verified retry behavior
- **Runtime Crash Testing**: Manually terminated healthy processes to test runtime restart logic
- **UI Event Testing**: Confirmed new events are properly emitted to frontend
- **Shutdown Testing**: Verified graceful shutdown during various stages of retry loops
- **Edge Case Testing**: Tested behavior with corrupted binaries, permission issues, and rapid shutdown scenarios

**What process can a PR reviewer use to test or verify this change?**
---

### **1. Binary Startup Retry Testing:**
```bash
# Temporarily rename a binary to trigger startup failures
mv minotari_node minotari_node.backup
# Start Universe and observe:
# - Console logs showing retry attempts (1/10, 2/10, etc.)
# - Frontend events for BinaryStartupAttempt
# - Final BinaryPermanentFailure after 10 attempts
# - Critical error dialog in UI
```

### **2. Runtime Restart Testing:**
```bash
# Start Universe normally, then kill a process:
kill -9 $(pgrep minotari_node)
# Observe:
# - Automatic detection of unhealthy process
# - Runtime restart attempts (1/3, 2/3, 3/3)
# - Process recovery or permanent failure notification
```

### **3. Configuration Verification:**
```rust
// Verify retry parameters can be customized in process_watcher.rs:
max_startup_attempts: 10,           // Initial startup retries
startup_retry_delay: 5s,            // Delay between startup attempts  
max_runtime_restart_attempts: 3,    // Runtime crash retries
runtime_restart_retry_delay: 10s,   // Delay between runtime attempts
```

**Breaking Changes**
---
- [x] None

### CODE REVIEW

File: src-tauri/src/process_watcher.rs [hunk 1]
```diff
@@ -142,7 +144,14 @@ impl<TAdapter: ProcessAdapter> ProcessWatcher<TAdapter> {
                         warn!(target: LOG_TARGET, "Process '{}' died immediately after start (ping failed).", name);
                         break;
                     }
-
+    
+                    // ADD THIS: Skip health checks during grace period
+                    if Instant::now() < grace_deadline {
```
[the remark this answers is not in the document]
Comment (fluffypony) [reply src-tauri/src/process_watcher.rs]: Good point - I modified it so that there's no grace period if a binary is instantly healthy

File: src-tauri/src/main.rs [hunk 2]
```diff
@@ -665,15 +769,17 @@ fn main() {
...
-
         match event {
         tauri::RunEvent::Ready  => {
             info!(target: LOG_TARGET, "RunEvent Ready");
             let handle_clone = app_handle.clone();
             tauri::async_runtime::spawn(async move {
                 SetupManager::get_instance().start_setup(handle_clone.clone()).await;
                 SetupManager::spawn_sleep_mode_handler(handle_clone.clone()).await;
+                // Initialize frontend updates after setup is complete
+                if let Err(e) = initialize_frontend_updates(&handle_clone).await {
```
[the remark this answers is not in the document]
Comment (fluffypony) [reply src-tauri/src/main.rs]: @Misieq01 good catch; fixed

**Comment by fluffypony** [comment]
Closed in favour of #2316
